var/spack/repos/builtin/packages/intel-mpi/package.py

```diff
@@ -62,7 +74,13 @@ def setup_dependent_package(self, module, dep_spec):
         else:
             raise RuntimeError('No suitable bindir found')
 
-        self.spec.mpicc = join_path(bindir, 'mpicc')
```

Comment:
> Also, it is confusing since the Intel-specific wrapper scripts will also obey the I_MPI_CC environment variables (i.e., running I_MPICC=gcc mpiicc would actually invoke the gcc compiler).

Thanks, that's new to me. I was pretty sure the compiler was fixed in those wrappers. In this case I agree it makes no sense changing them.